src/procfile.h

```diff
@@ -27,86 +28,164 @@
 
 
 // ----------------------------------------------------------------------------
 // The procfile
 
-#define PROCFILE_FLAG_DEFAULT             0x00000000
-#define PROCFILE_FLAG_NO_ERROR_ON_FILE_IO 0x00000001
+#define PROCFILE_FLAG_DEFAULT             0x00000000 ///< default flag
+#define PROCFILE_FLAG_NO_ERROR_ON_FILE_IO 0x00000001 ///< do not fail/error on file io
 
+/** The procfile */
 typedef struct {
-    char filename[FILENAME_MAX + 1];
-    uint32_t flags;
-    int fd;         // the file desriptor
-    size_t len;     // the bytes we have placed into data
-    size_t size;        // the bytes we have allocated for data
-    pflines *lines;
-    pfwords *words;
-    char separators[256];
-    char data[];        // allocated buffer to keep file contents
+    char filename[FILENAME_MAX + 1]; ///< the filename
+    uint32_t flags;       ///< PROCFILE_FLAG_*
+    int fd;               ///< the file desriptor
+    size_t len;           ///< the bytes we have placed into data
+    size_t size;          ///< the bytes we have allocated for data
+    pflines *lines;       ///< lines of the file
+    pfwords *words;       ///< words of the file
+    char separators[256]; ///< sparator chars
+    char data[];          ///< allocated buffer to keep file contents
 } procfile;
 
-// close the proc file and free all related memory
+/**
+ * Close the proc file and free all related memory.
+ *
+ * @param ff File to close.
+ */
 extern void procfile_close(procfile *ff);
 
-// (re)read and parse the proc file
+/**
+ * Parse an open procfile.
+ *
+ * (re)read and parse the proc file
+ *
+ * @param ff File to parse.
+ * @return The parsed file. `NULL` on error.
+ */
 extern procfile *procfile_readall(procfile *ff);
 
-// open a /proc or /sys file
+/**
+ * Open a /proc or /sys file.
+ *
+ * @param filename to open
+ * @param separators used to seperate columns
+ * @param flags PROCFILE_FLAG_*
+ * @return the opend procfile. `NULL` on error.
+ */
 extern procfile *procfile_open(const char *filename, const char *separators, uint32_t flags);
 
-// re-open a file
-// if separators == NULL, the last separators are used
+/**
+ * re-open a /proc or /sys file.
+ *
+ * If `separators == NULL`, the last separators are used
+ *
+ * @param ff Closed procfile.
+ * @param filename to re-open
+ * @param separators used to seperate columns or `NULL`
+ * @param flags PROCFILE_FLAG_*
+ * @return the opend procfile. `NULL` on error.
+ */
 extern procfile *procfile_reopen(procfile *ff, const char *filename, const char *separators, uint32_t flags);
 
-// example walk-through a procfile parsed file
+/**
+ * Example walk-through a procfile parsed file.
+ *
+ * @param ff File to print.
+ */
 extern void procfile_print(procfile *ff);
 
+/** 
+ * Add characters in quotes to the set of quotes.
+ *
+ * @param ff Proc file.
+ * @param quotes Characters to set as quote.
+ */
 extern void procfile_set_quotes(procfile *ff, const char *quotes);
+/**
+ * ktsaou: Your help needed
```

Comment:
procfile supports quotes and opening / closing characters.
So, if you set this to `(` and `)`, the tokenizer will assume everything in parenthesis as a single keyword.